RecoEgamma/EgammaPhotonAlgos/src/ConversionHitChecker.cc

```diff
@@ -16,21 +16,23 @@ std::pair<uint8_t,Measurement1DFloat> ConversionHitChecker::nHitsBeforeVtx(const
 
   //iterate inside out, when distance to vertex starts increasing, we are at the closest hit
   // the first (and last, btw) hit is always valid... (apparntly not..., conversion is different????)
-  auto hb = track.recHitsBegin();
-  unsigned int ih=0;
-  for(;ih<track.recHitsSize();++ih) {
-    auto recHit = *(hb+ih);
-    if (recHit->isValid()) break; 
+  TrackingRecHit const* recHit = nullptr;
+  unsigned int closest = 0;
+  for(auto const& hit : track.recHits()) {
+    if (hit->isValid()) {
+      recHit = hit;
+      break; 
+    }
+    ++closest;
   }
-  auto recHit = *(hb+ih);
-  unsigned int closest=ih;
+  if(recHit == nullptr) throw cms::Exception("No valid RecHit found in track!");
```

Comment:
IIUC, this changes the logic.
In the case of all invalid hits the original version would pick the first hit,
while this update will throw.
I can't recall if we actually have such cases.
I prefer not to change the algorithm behavior in otherwise purely technical updates.